twake/backend/node/src/services/channels/services/member/service.ts

```diff
@@ -583,17 +618,32 @@ export class Service implements MemberService {
     return String(member.user_id) === String(user.id);
   }
 
-  isChannelMember(user: User, channel: Channel): Promise<ChannelMember> {
+  async isChannelMember(
+    user: User,
+    channel: Partial<Pick<Channel, "company_id" | "workspace_id" | "id">>,
+    cacheTtlSec?: number,
+  ): Promise<ChannelMember> {
     if (!user) {
       return;
     }
 
-    return this.get({
-      channel_id: channel.id,
-      company_id: channel.company_id,
-      workspace_id: channel.workspace_id,
-      user_id: user.id,
-    });
+    const get = () =>
+      this.get({
+        channel_id: channel.id,
+        company_id: channel.company_id,
+        workspace_id: channel.workspace_id,
+        user_id: user.id,
+      });
```

Comment:
I don't think we still need a function in a function, you can move this code block to the end return (the variable `const get` is only used once